services/question-service/entry-points/api/routes.ts

```diff
@@ -0,0 +1,63 @@
+import express from 'express';
+import * as questionUseCase from '../../domain/question-use-case';
+import { validateAddQuestionInput, validateUpdateQuestionInput } from './validators';
+
+export default function defineRoutes(expressApp: express.Application) {
+  const router = express.Router();
+
+  router.put('/',validateAddQuestionInput ,async (req, res, next) => {
+    try {
+      const addQuestionResponse = await questionUseCase.addQuestion(req.body);
+      return res.json(addQuestionResponse);
+    } catch (error) {
+      next(error);
+    }
+  });
+
+  router.get('/', async (_, res, next) => {
+    try {
+      const response = await questionUseCase.getAllQuestions();
+      res.json(response);
+    } catch (error) {
+      next(error);
+    }
+  })
+
+  router.get('/id/:id', async (req, res, next) => {
+    try {
+      const response = await questionUseCase.getQuestionById(req.params.id);
+      res.status(200).json(response);
+    } catch (error) {
+      next(error);
+    }
+  });
+
+  router.get('/title/:title', async (req, res, next) => {
+    try {
+      const response = await questionUseCase.getQuestionByTitle(req.params.title);
+      res.status(202).json(response);
+    } catch (error) {
+      next(error)
+    }
+  });
```

Comment:
To align with RESTful conventions, I suggest just keeping the `router.get('/')` route, and getting any query parameters there and using them to query the database. So the one route would allow us to handle all different types of queries, rather than needing one controller per query type.